src/quo2/components/settings/data_item/component_spec.cljs

```diff
@@ -0,0 +1,158 @@
+                :description :account
+                :icon-right? false
+                :card?       true
+                :label       :none
+                :status      :default
+                :size        :default
+                :title       "Label"
+                :subtitle    "Description"
+                :icon        :i/placeholder}])
+    (h/is-truthy (h/get-by-text "Description")))
+
+  (h/test "right icon is not visible when icon-right? is false"
+    (h/render [quo/data-item
+               {:on-press    #(js/alert (str "pressed"))
+                :blur?       false
+                :description :account
+                :icon-right? false
+                :card?       true
+                :label       :none
+                :status      :default
+                :size        :default
+                :title       "Label"
+                :subtitle    "Description"
+                :icon        :i/placeholder}])
+    (h/is-falsy (h/query-by-label-text :icon-right)))
+
+  (h/test "right icon is visible when icon-right? is true"
+    (h/render [quo/data-item
+               {:on-press    #(js/alert (str "pressed"))
+                :blur?       false
+                :description :account
+                :icon-right? true
+                :card?       true
+                :label       :none
+                :status      :default
+                :size        :default
+                :title       "Label"
+                :subtitle    "Description"
+                :icon        :i/placeholder}])
+    (h/is-truthy (h/query-by-label-text :icon-right)))
+
+  (h/test "description icon is visible when description is icon"
+    (h/render [quo/data-item
+               {:on-press    #(js/alert (str "pressed"))
+                :blur?       false
+                :description :icon
+                :icon-right? true
+                :card?       true
+                :label       :preview
+                :status      :default
+                :size        :default
+                :title       "Label"
+                :subtitle    "Description"
+                :icon        :i/placeholder}])
+    (h/is-truthy (h/query-by-label-text :description-icon)))
+
+  (h/test "description image is visible when description is network"
+    (h/render [quo/data-item
+               {:on-press    #(js/alert (str "pressed"))
+                :blur?       false
+                :description :network
+                :icon-right? true
+                :card?       true
+                :label       :preview
+                :status      :default
+                :size        :default
+                :title       "Label"
+                :subtitle    "Description"
+                :icon        :i/placeholder}])
+    (h/is-truthy (h/query-by-label-text :description-image)))
+
+  (h/test "description emoji is visible when description is account"
+    (h/render [quo/data-item
+               {:on-press    #(js/alert (str "pressed"))
```

Comment:
we should use mock events for on-press. 
Also for tests it's best to use the minimum amount of data required to make the test work imo

Reply:
Done